internal/server/http/server_test.go

```diff
@@ -210,3 +210,54 @@ func TestBuildSecureMode(t *testing.T) {
 		})
 	}
 }
+
+func TestPrepareAccessControl(t *testing.T) {
```

Comment:
In addition to this, could you please write an "integration" test to make sure Killgrave behaves as expected? I'd also help understanding why we need it.

Thanks! 🙇🏻